RELEASE.md

```diff
@@ -0,0 +1,43 @@
+# Releasing connect-go
+
+This document outlines how to create a release of connect-go.
+
+1. Clone the repo, ensuring you have the latest main.
+
+2. On a new branch, open [connect.go](connect.go) and change the `Version` constant. Do not just remove the `-dev` suffix: look at the release history and the unreleased commits to choose a new semantic version number.
```

Comment:
This confused me: with the version at `1.14.0-dev` we should have chosen a patch release of `1.13.1` instead of the `1.14.0` release as no API changes were made. I think we need to explain the reasoning around why we default to increasing the minor first over increasing on removing the suffix. As an aside it's more difficult to know what the current version is without going through the git history (checking releases). In contrast protobuf-go will use append the suffix without incrementing: https://github.com/protocolbuffers/protobuf-go/commit/9f73929456f40461a01f2375d7dbf875126bd6ce . What's the benefit over incrementing the minor with the dev suffix for anticipating the release?

Reply:
My suggestion would be:
1. Update the wording here such that you wouldn't have been confused given the historical and current behavior.
2. After we land this PR, we can debate whether to change the practice. FWIW, my instinct would be to do what protobuf-go does for a new repo, but that is something for maintainers to debate/decide.

Reply:
> Check for MINOR or PATCH changes. On MINOR confirm the current version by removing the `-dev` suffix. On PATCH look through the release history and increment the latest patch of the previous minor release.

Reply:
What do you think of this? I would have also liked to include a sample diff for each case, but I can't do that as a GitHub suggestion since it would require putting a \``` block inside of a \``` block
```suggestion
2. On a new branch, open [connect.go](connect.go) and change the `Version` constant to an appropriate [semantic version](https://semver.org/) given the changes that are being released.
  * If there are only bug fixes and no new features, remove the `-dev` suffix and set the version number equal to the latest release with the PATCH component incremented by 1.
  * If the new release includes features, remove the `-dev` suffix, set the MINOR number to be 1 more than the MINOR number of the latest release, and zero out the PATCH number. In the common case, the diff here will just be to remove the `-dev` suffix.
```

Reply:
I think it's important to highlight the latest release is not the current one in the `Version` name.  `to the latest release with the PATCH component incremented by 1` should say something about finding the release like the current `look at the release history`.

Reply:
what about this?
```suggestion
2. On a new branch, open [connect.go](connect.go) and change the `Version` constant to an appropriate [semantic version](https://semver.org/). To select the correct version, look at the version number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest) and the changes that are included in this new release.
  * If there are only bug fixes and no new features, remove the `-dev` suffix, set MINOR number to be equal to the [latest release](https://github.com/connectrpc/connect-go/releases/latest), and set the PATCH number to be 1 more than the PATCH number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest).
  * If there are features being released, remove the `-dev` suffix, set the MINOR number to be 1 more than the MINOR number of the [latest release](https://github.com/connectrpc/connect-go/releases/latest), and set the PATCH number to `0`. In the common case, the diff here will just be to remove the `-dev` suffix.
```